source/tests/system/nirfmxlte_driver_api_tests.cpp

```diff
@@ -85,6 +102,1192 @@ TEST_F(NiRFmxLTEDriverApiTests, InitializeFromNIRFSA_Close_Succeeds)
+  EXPECT_SUCCESS(session, client::obw_cfg_averaging(stub(), session, "", OBW_AVERAGING_ENABLED_FALSE, 10, OBW_AVERAGING_TYPE_RMS));
+  EXPECT_SUCCESS(session, client::initiate(stub(), session, "", ""));
+
+  const auto obw_fetch_spectrum_response = EXPECT_SUCCESS(session, client::obw_fetch_spectrum(stub(), session, "", 10.0));
+  const auto obw_fetch_measurement_response = EXPECT_SUCCESS(session, client::obw_fetch_measurement(stub(), session, "", 10.0));
+
+  EXPECT_LT(0.0, obw_fetch_spectrum_response.x0());
+  EXPECT_LT(0.0, obw_fetch_spectrum_response.dx());
+  EXPECT_EQ(403, obw_fetch_spectrum_response.spectrum_size());
+  EXPECT_EQ(403, obw_fetch_spectrum_response.spectrum().size());
+  EXPECT_GT(0.0, obw_fetch_spectrum_response.spectrum(0));
+  EXPECT_LT(0.0, obw_fetch_measurement_response.occupied_bandwidth());
+  EXPECT_LT(0.0, obw_fetch_measurement_response.absolute_power());
+  EXPECT_LT(0.0, obw_fetch_measurement_response.start_frequency());
+  EXPECT_LT(0.0, obw_fetch_measurement_response.stop_frequency());
+}
+
+TEST_F(NiRFmxLTEDriverApiTests, SEMContiguousMultiCarrierFromExample_FetchData_DataLooksReasonable)
+{
+  const auto NUMBER_OF_COMPONENT_CARRIERS = 2;
+  int32 linkDirection = LINK_DIRECTION_UPLINK;
+  const std::vector<int> empty_int_vec;
+  std::vector<float64> componentCarrierBandwidth{20e6, 20e6};
+  std::vector<float64> componentCarrierFrequency{-9.9e6, 9.9e6};
+  std::vector<float64> componentCarrierMaximumOutputPower{0.0, 0.0};
+  auto session = init_session(stub(), PXI_5663E);
+  EXPECT_SUCCESS(session, client::cfg_frequency_reference(stub(), session, "", FREQUENCY_REFERENCE_SOURCE_ONBOARD_CLOCK, 10.0e6));
+  EXPECT_SUCCESS(session, client::cfg_rf(stub(), session, "", 1.95e9, 0.0, 0.0));
+  EXPECT_SUCCESS(session, client::cfg_digital_edge_trigger(stub(), session, "", DIGITAL_EDGE_TRIGGER_SOURCE_PFI0, DIGITAL_EDGE_TRIGGER_EDGE_RISING, 0.0, false));
+  EXPECT_SUCCESS(session, client::cfg_component_carrier_spacing(stub(), session, "", COMPONENT_CARRIER_SPACING_TYPE_NOMINAL, -1));
+  EXPECT_SUCCESS(session, client::cfg_number_of_component_carriers(stub(), session, "", NUMBER_OF_COMPONENT_CARRIERS));
+  EXPECT_SUCCESS(session, client::cfg_component_carrier_array(stub(), session, "", componentCarrierBandwidth, componentCarrierFrequency, empty_int_vec));
+  EXPECT_SUCCESS(session, client::cfg_link_direction(stub(), session, "", LINK_DIRECTION_UPLINK));
+  EXPECT_SUCCESS(session, client::select_measurements(stub(), session, "", MEASUREMENT_TYPES_SEM, true));
+  EXPECT_SUCCESS(session, client::sem_cfg_sweep_time(stub(), session, "", SEM_SWEEP_TIME_AUTO_TRUE, 0.001));
+  EXPECT_SUCCESS(session, client::sem_cfg_averaging(stub(), session, "", SEM_AVERAGING_ENABLED_FALSE, 10, SEM_AVERAGING_TYPE_RMS));
+  if (linkDirection == LINK_DIRECTION_UPLINK) {
+    EXPECT_SUCCESS(session, client::sem_cfg_uplink_mask_type(stub(), session, "", SEM_UPLINK_MASK_TYPE_GENERAL_NS01));
+  }
+  else if (linkDirection == LINK_DIRECTION_DOWNLINK) {
```

Comment:
Can you remove `linkDirection` and either keep the body from the `if` or the `else if`? We've been trying to remove the unexecuted code blocks in the tests.